mcpwm_foc.c

```diff
@@ -2856,12 +2847,13 @@ void mcpwm_foc_adc_int_handler(void *p, uint32_t flags) {
 						conf_now->foc_sl_erpm_hfi,
 						motor_now);
 
-				if (!motor_now->m_phase_override) {
-					id_set_tmp = 0.0;
-				}
 				break;
 			}
 
+            if (!motor_now->m_phase_override & !(motor_now->m_control_mode == CONTROL_MODE_OPENLOOP_PHASE)) {
```

Comment:
I feel like this formulation is brittle. If a new control mode is introduced, it can lead to `id_set_tmp` being set by accident. It might make sense to write this as a switch..case:

```
switch(motor_now->m_control_mode)
case FOC_SENSOR_MODE_HALL:
case FOC_SENSOR_MODE_SENSORLESS:
case FOC_SENSOR_MODE_HFI:
case FOC_SENSOR_MODE_HFI_START:
  if (!motor_now->m_phase_override) {
    id_set_tmp = 0.0;
  }
  break;
default:
```

The reason to use a switch is because the compiler can provide a safety that all control modes are covered.